src/circuit/gadget/ecc/chip/mul_fixed/short.rs

```diff
@@ -357,4 +369,144 @@ pub mod tests {
+            utilities::{CellValue, UtilitiesInstructions},
+        };
+        use halo2::{
+            circuit::{Layouter, SimpleFloorPlanner},
+            dev::{MockProver, VerifyFailure},
+            plonk::{Circuit, ConstraintSystem, Error},
+        };
+
+        #[derive(Default)]
+        struct MyCircuit {
+            magnitude: Option<pallas::Base>,
+            sign: Option<pallas::Base>,
+        }
+
+        impl UtilitiesInstructions<pallas::Base> for MyCircuit {
+            type Var = CellValue<pallas::Base>;
+        }
+
+        impl Circuit<pallas::Base> for MyCircuit {
+            type Config = EccConfig;
+            type FloorPlanner = SimpleFloorPlanner;
+
+            fn without_witnesses(&self) -> Self {
+                Self::default()
+            }
+
+            fn configure(meta: &mut ConstraintSystem<pallas::Base>) -> Self::Config {
+                let advices = [
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                    meta.advice_column(),
+                ];
+
+                let constants = [meta.fixed_column(), meta.fixed_column()];
+                let perm = meta.permutation(
+                    &advices
+                        .iter()
+                        .map(|advice| (*advice).into())
+                        .chain(constants.iter().map(|fixed| (*fixed).into()))
+                        .collect::<Vec<_>>(),
+                );
+
+                let lookup_table = meta.fixed_column();
+                EccChip::configure(meta, advices, lookup_table, constants, perm)
+            }
+
+            fn synthesize(
+                &self,
+                config: Self::Config,
+                mut layouter: impl Layouter<pallas::Base>,
+            ) -> Result<(), Error> {
+                let column = config.advices[0];
+
+                let short_config: super::Config = (&config).into();
+                let magnitude_sign = {
+                    let magnitude = self.load_private(
+                        layouter.namespace(|| "load magnitude"),
+                        column,
+                        self.magnitude,
+                    )?;
+                    let sign =
+                        self.load_private(layouter.namespace(|| "load sign"), column, self.sign)?;
+                    (magnitude, sign)
+                };
+
+                short_config.assign(layouter, magnitude_sign, &ValueCommitV::get())?;
+
+                Ok(())
+            }
+        }
+
+        // Magnitude larger than 64 bits should fail
+        {
+            let circuit = MyCircuit {
+                magnitude: Some(pallas::Base::from_u128(1 << 64)),
```

Comment:
Test this for `1<<66` and `1<<254` as well.

Reply:
I realise that `1 << 66` would give the same failure as `1 << 64`, because the "last window check" is on the running sum, which will be nonzero (even if the three-bit window itself is zero).